en/resources/tutorials/templating.md

```diff
@@ -31,36 +29,83 @@ You can provide your own Kitura template engine by implementing the `TemplateEng
+2. Create a directory  
+`mkdir KituraTemplateExample`
+3. Change to the new directory  
+`cd KituraTemplateExample`
+4. Create the Kitura server  
+`kitura init`
 
-To use a template engine you need to specify a dependency in `Package.swift` for the engine you want to use. You can specify multiple dependencies if you wish to support more than one template engine (e.g. both Mustache and Stencil). In our example we'll add both templates to demonstrate GRMustache and Stencil integration.
+This will take a couple of minutes to complete, `kitura init` downloads Kitura and various supporting packages for monitoring the status of your server and configuration files to allow you to easily deploy to a Docker container, a Kubernetes cluster, or the IBM Cloud. 
 
+## Adding a template engine to your Kitura app
+Kitura is a lightweight, modular framework &mdash; this means that by default you get only basic HTTP functionality, like routing, parameter handling, parsing request body, etc. You can augment Kitura's out-of-the-box functionality by adding additional packages using the [Swift Package Manager](https://swift.org/package-manager/), or [SPM](https://swift.org/package-manager/) for short. The `kitura init` command we ran earlier added some of these packages for you already, we need to add two more. The two packages we'd like to add are the template engines we'd like to use: 
+
+* Kitura-StencilTemplateEngine  
+`.package(url: "https://github.com/IBM-Swift/Kitura-StencilTemplateEngine.git", .upToNextMajor(from: "1.0.0"))`
+* Kitura-MustacheTemplateEngine  
+`.package(url: "https://github.com/IBM-Swift/Kitura-MustacheTemplateEngine.git", .upToNextMajor(from: "1.0.0"))`
+### Adding the template engine packages
+We're going to add both template engines to our Kitura app, to do this we need to add the packages to our `Package.swift` file (this was also generated by `kitura init`). In your terminal run the following: 
+
+1. Ensure we're in the correct directory  
+`cd ~/Desktop/KituraTemplateExample`
+
+2. Open the `Package.swift` file  
+`open Package.swift`
+
+3. Add the packages to the list of dependecies, so below the line: 
+`.package(url: "https://github.com/IBM-Swift/Health.git", from: "0.0.0"),`
+Add the following:  
+`.package(url: "https://github.com/IBM-Swift/Kitura-MustacheTemplateEngine.git", .upToNextMajor(from: "1.0.0")),`  
+ `.package(url: "https://github.com/IBM-Swift/Kitura-StencilTemplateEngine.git", .upToNextMajor(from: "1.0.0")),`
+
+4. Add the template engines to the correct `target`. At the end of the line:  
+`.target(name: "Application", dependencies: [ "Kitura", "Configuration", "CloudEnvironment","SwiftMetrics","Health", ]),`  
+Add `"KituraStencil", "KituraMustache"`  
+The result should look like this:   
+`.target(name: "Application", dependencies: [ "Kitura", "Configuration", "CloudEnvironment","SwiftMetrics","Health", "KituraStencil", "KituraMustache"]),`  
+
+After those changes our `Package.swift` should look like this:
```

Comment:
I wanted this to be a tutorial where a user wouldn't need to keep jumping to different pages to run through unless they wanted more information on something, like Codable for example. 

I mentioned the following idea to you: 
I think the instructions of how to setup your Package.swift should exist in the README of the project's repo rather than every tutorial having to explain this step.